test/unit/three_scale/json_validator_test.rb

```diff
@@ -35,7 +35,7 @@ class ThreeScale::JSONValidatorTest < ActiveSupport::TestCase
     json = { json_key: 'json_value' }
     schema = {}
     validator = @klass.new(json)
-    @klass.expects(:fully_validate).with(schema, json, any_parameters)
+    ::JSON::Validator.expects(:fully_validate).with(schema, json, any_parameters)
```

Comment:
I believe that this test passed in the past because of some mocha 1.1.0 bug.

It only passed if run after `autoload schema files` test. But still it would not pass if any one or both of these two lines from the `autoload schema files` test were commented out
```rb
    ::JSON::Validator.expects(:add_schema).times(schema_files_count)
    @klass.autoload_schemas
```

I tried updating mocha to 2.1.0 and after the update, the test would never pass regardless of test execution order. So I believe that the test should just test for `::JSON::Validator.expects(..` because `ThreeScale::JSONValidator` itself calls `::JSON::Validator.fully_validate(schema, json, options)`. And I couldn't find any possible way for `@klass.fully_validate` to be called by it.

cc @jlledom